gpu/gpu_info_cuda.c

```diff
@@ -0,0 +1,110 @@
+#ifndef __APPLE__  // TODO - maybe consider nvidia support on intel macs?
+
+#include "gpu_info_cuda.h"
+
+#include <string.h>
+
+#ifndef _WIN32
+const char *cuda_lib_paths[] = {
+    "libnvidia-ml.so",
+    "/usr/local/cuda/lib64/libnvidia-ml.so",
+    NULL,
+};
+#else
+const char *cuda_lib_paths[] = {
+    "nvml.dll",
+    "",
+    NULL,
+};
+#endif
+
+void cuda_init(cuda_init_resp_t *resp) {
+  resp->err = NULL;
+  const int buflen = 256;
+  char buf[buflen + 1];
+  int i;
+
+  struct lookup {
+    char *s;
+    void **p;
+  } l[4] = {
+      {"nvmlInit_v2", (void *)&resp->ch.initFn},
```

Comment:
I think this might break `wsl` which a lot of people are currently using to run Ollama on Windows. The GPU is accessible to WSL (I can see it by running `nvidia-smi`), but I see the error `CUDA not detected: Unable to load libnvidia-ml.so library to query for nvidia GPUs: ... : no such file or directory` in the logs. This file on the host system wont be accessible in WSL but the GPU will still work.

Reply:
I think using `nvidia-smi` as a fallback could be a good solution.

Reply:
Hmm... that's odd.  I would have thought that if nvidia-smi was inside the WSL vhd image that it would also have the underlying library that CLI depends on, which is what I'm tying to load.  Maybe we do have to fallback to CLI, but let me poke around a bit and see if there's another way...

Reply:
I've tried a couple possibilities, but no solution yet...

Turns out there's a `/usr/local/cuda/lib64/stubs/libnvidia-ml.so` in WSL2 if you have the CUDA libs installed, but at runtime that fails with a wall of text error and code:
```
NVML_ERROR_DRIVER_NOT_LOADED = 9
NVIDIA driver is not loaded.
```

I also found `/mnt/c/Windows/System32/lxss/lib/libnvidia-ml.so.1` is present, however that also fails at runtime with an init error of:
```
NVML_ERROR_OPERATING_SYSTEM = 17
The GPU control device has been blocked by the operating system/cgroups.
```
(root or normal user doesn't matter)

That last error feels like it might be solvable if we can figure out how things are wired up, but I'm not sure.

My inclination would be to merge what we've got and note this as a limitation on WSL2 for the pre-release (Your call @jmorganca ). I imagine a lot of people running in WSL2 will be happier with it running natively, which it does in its current form.  We can continue to investigate if there's a viable solution to get libnvidia-ml.so to load inside WSL2, and if not, fallback to the CLI as @BruceMacD suggested.

Reply:
Scratch that - I mistakenly tested under wsl1 which won't work.  Reset my setup to wsl2 and found a fix.

There's probably room to improve detection and error reporting for misconfigurations.